### DISCUSSION TITLE
How to create Organisation Secrets in Mend Developer Platform for Bitbucket Cloud

### DISCUSSION THREAD

**Discussion by mcr-paulanand** [renovatebot/renovate#30440] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

None

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Mend Developer Platform (developer.mend.io)

### Please tell us more about your question or problem

Hi,

**Background**
- I have managed to set up integration between Bitbucket Cloud and Mend Developer Platform.
- I have added a repository secret to store the github.com API token for changelog retrieval.

**Problem**
- I am trying to store the github.com API token as an organisation secret rather than per-repo
- I am unable to find the page to configure an organisation secret.
- How can I setup this secret as an organisation secret?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
https://docs.renovatebot.com/getting-started/app-secrets/#managing-secrets-for-the-mend-hosted-cloud-apps

**Comment by mcr-paulanand** [comment]
- I am unable to see the settings button for my workspace.
- I am using BitBucket Cloud and I am the Workspace Admin.
- I am able to see the settings for individual repos, just not the workspace.

![image](https://github.com/user-attachments/assets/78d1d972-3020-40cd-8fdb-638dabfae596)

**Reply by rarkins** [reply]
The Settings button should show up if you're Workspace admin. We'll see if we can reproduce this problem.

**Comment by mcr-paulanand** [comment]
Update:
- When using my personal Bitbucket Cloud account, the settings button is available.
- However, when using my Organisation account, it is not available.
- I tried reinstalling the Mend addon in Bitbucket Cloud but it is still the same.

**Reply by rarkins** [reply]
I also see the Settings button for my own account: 
<img width="515" alt="image" src="https://github.com/user-attachments/assets/9f3f4703-9e07-4930-a630-07c7b8c51f1a">

I'm not sure how to replicate the scenario you have. Did you create the Workspace, or were you added to it by someone who did?

**Reply by mcr-paulanand** [reply]
I created the Workspace. I am a Workspace Admin.

![image](https://github.com/user-attachments/assets/a2f358a9-132c-499f-915d-0ef8c2f0ba48)

Is there a way I can reset the account in the Mend Developer Platform?

Maybe I can reconfigure the whole thing and see if it works.

**Reply by rarkins** [reply]
There isn't really a reset option. Permissions aren't static though (e.g. decided at time of installation). Instead the backend uses the OAuth token per request to determine access rights.

You could try going directly to `https://developer.mend.io/bitbucket/<workspace>/-/settings` in your browser too.

I just installed into an old Workspace and was able to see Settings successfully, so the problem you're having cannot be as simple as "Workspace settings don't work". I suggest you try creating a new Workspace, installing Mend into that, and seeing if that one works for settings or not too.

**Reply by rarkins** [reply]
This is what I see on the Workspace which works for me:

<img width="950" alt="image" src="https://github.com/user-attachments/assets/f114fd11-351e-44d3-bd53-bd7574363cf8">

**Reply by mcr-paulanand** [reply, marked as answer]
Found the issue. I think it has something to do with IP whitelisting being turned on.

**Workaround**
- Disable IP whitelisting
- Login to Mend Developer Platform -> Able to see Settings button
- Logout from Mend Developer Platform
- Reenable IP whitelisting 
- Login to Mend Developer Platform -> Unable to see Settings button

**Reply by rarkins** [reply]
Would it help if we gave you the static IPs of the service? These might change in future but for now there's two static IPs which all requests come through. I'm not sure if there's a way to configure them in the Bitbucket app settings themselves

**Reply by mcr-paulanand** [reply]
Yes, this would help.

We can add the IPs in Bitbucket.

![image](https://github.com/user-attachments/assets/ff61bb84-121a-4b4d-bc91-6358a0c62a95)

**Reply by rarkins** [reply]
Try adding these two:

34.239.12.110
3.222.82.38

**Reply by mcr-paulanand** [reply]
- Adding these two IPs did not help.
- So far, only disabling IP whitelisting allowed me to change the Organisation settings.
- As a workaround, I have disabled IP whitelisting, made whatever changes and then reenabled IP whitelisting.
